src/internal.c

```diff
@@ -4726,31 +4726,48 @@ static int DoUserAuthRequestEcc(WOLFSSH* ssh, WS_UserAuthData_PublicKey* pk,
     }
 
     if (ret == WS_SUCCESS) {
-#ifdef WOLFSSH_SMALL_STACK
-    key_ptr = (ecc_key*)WMALLOC(sizeof(ecc_key), ssh->ctx->heap,
-            DYNTYPE_PUBKEY);
-    sig_r_ptr = (mp_int*)WMALLOC(sizeof(mp_int), ssh->ctx->heap, DYNTYPE_MPINT);
-    sig_s_ptr = (mp_int*)WMALLOC(sizeof(mp_int), ssh->ctx->heap, DYNTYPE_MPINT);
-    if (key_ptr == NULL || sig_r_ptr == NULL || sig_s_ptr == NULL)
-        ret = WS_MEMORY_E;
-#else
-    key_ptr = &s_key;
-    sig_r_ptr = &sig_r;
-    sig_s_ptr = &sig_s;
-#endif
+    #ifdef WOLFSSH_SMALL_STACK
+        key_ptr = (ecc_key*)WMALLOC(sizeof(ecc_key), ssh->ctx->heap,
+                DYNTYPE_PUBKEY);
+        if (key_ptr == NULL) {
+            ret = WS_MEMORY_E;
+        }
+    #else
+        key_ptr = &s_key;
+    #endif
     }
 
-    if (sig_r_ptr != NULL) {
+    if (ret == WS_SUCCESS) {
+    #ifdef WOLFSSH_SMALL_STACK
+        sig_r_ptr = (mp_int*)WMALLOC(sizeof(mp_int), ssh->ctx->heap, DYNTYPE_MPINT);
+        if (sig_r_ptr == NULL) {
+            ret = WS_MEMORY_E;
+        }
+        else
+    #else
+        sig_r_ptr = &sig_r;
+    #endif
         if (mp_init(sig_r_ptr) != MP_OKAY) {
+            sig_r_ptr = NULL;
             ret = WS_MEMORY_E;
         }
     }
-    if (sig_s_ptr != NULL) {
+    if (ret == WS_SUCCESS) {
+    #ifdef WOLFSSH_SMALL_STACK
+        sig_s_ptr = (mp_int*)WMALLOC(sizeof(mp_int), ssh->ctx->heap, DYNTYPE_MPINT);
+        if (sig_s_ptr == NULL) {
+            ret = WS_MEMORY_E;
+        }
+        else
+    #else
+        sig_s_ptr = &sig_s;
+    #endif
         if (mp_init(sig_s_ptr) != MP_OKAY) {
+            sig_s_ptr = NULL;
```

Comment:
should sig_s_ptr be free'd in the WOLFSSH_SMALL_STACK case before set to NULL?

Reply:
Yep, that's right... Pushed update.